site/src/pages/release-notes/1.23.0.mdx

```diff
@@ -0,0 +1,268 @@
+  ```java
+  ServerBuilder sb = ...
+  sb.requestIdGenerator((routingCtx, req) -> {
+    return RequestId.of(routingCtx.headers().getLong("requestId"));
+  });
+  ```
+- You can now collect metrics of open and closed connections
+  using <type://ConnectionPoolListener#metricCollecting(MeterRegistry)> on the client side. #4685 #4686
+  ```java
+  MeterRegistry registry = ...
+  ConnectionPoolListener listener = ConnectionPoolListener.metricCollecting(registry);
+  ClientFactory factory = ClientFactory.builder()
+                                       .connectionPoolListener(listener)
+                                       .build();
+  ```
+- <type://ClientRequestContext#authority()> can now be used to get the authority that will be sent to
+  the server. #4697
+  ```java
+  ClientRequestContext ctx = ...
+  String authority = ctx.authority();
+  ```
+- You can now use `ManagementServerProperties` to configure the management server when using
+  Spring integration. #4560 #4574
+  ```yaml
+  armeria:
+    ports:
+      - port: 8080
+  # Use different port with a base path:
+  management:
+    server:
+      port: 8443
+      base-path: /foo
+    endpoints:
+      web:
+        exposure:
+          include: health, loggers, prometheus
+  ```
+- You can now execute gRPC `ServerInterceptor` asynchronously in Kotlin
+  using <type://CoroutineServerInterceptor>. #4669 #4724
+  ```kotlin
+  class AuthInterceptor : CoroutineServerInterceptor {
+    private val authorizer = ...
+
+    override suspend fun <ReqT, RespT> suspendedInterceptCall(
+      call: ServerCall<ReqT, RespT>,
+      headers: Metadata,
+      next: ServerCallHandler<ReqT, RespT>
+    ): ServerCall.Listener<ReqT> {
+      val result = authorizer.authorize(ServiceRequestContext.current(), headers).await()
+      if (result) {
+        return next.startCall(call, headers)
+      } else {
+        throw AnticipatedException("Invalid access")
+      }
+    }
+  }
+  ```
+
+## 📈 Improvements
+
+- The debug page of <type://DocService> is now a pop-up for easier access. #4599
+- You can now use `Connection: close` header to close a connection after receiving a response. #4131 #4454 #4471 #4531
+- Armeria now exports the gauge metrics of <type://CommonPools#workerGroup()>. #4675 #4750
+  - The metric names are:
+    `armeria.netty.common.event.loop.workers` and `armeria.netty.common.event.loop.pending.tasks`
+- Armeria server now automatically routes HEAD requests to the GET method binding service on the same path. #4038 #4706
+
+## 🛠️ Bug fixes
+
+- The <type://RetryConfig> from <type://RetryConfigMapping> is retrieved only once for a retry session
+  and reused. #4753 #4778
+- Fixed a memory leak of the response body for HEAD requests. #4771
+- `#` character is now always percent-encoded in both path and query string. #4751 #4765
+- Self-suppression exception is not raised anymore in HTTP/1.1 encoder. #4749 #4758
```

Comment:
I think it would be nice to wrap in code blocks to denote that this isn't plain english

```suggestion
- `Self-suppression` exception is not raised anymore in HTTP/1.1 encoder. #4749 #4758
```